File: app-aws-lambda-example-go/containers/AppExampleLambdaFunction.go
package containers

import (
	"github.com/exproletariy/pip-services3-containers-examples/app-aws-lambda-example-go/build"
	cproc "github.com/pip-services3-go/pip-services3-aws-go/container"
)

type AppExampleLambdaFunction struct {
	cproc.CommandableLambdaFunction
}

func NewAppExampleLambdaFunction() *AppExampleLambdaFunction {
	c := AppExampleLambdaFunction{}
	c.CommandableLambdaFunction = *cproc.NewCommandableLambdaFunction("app-example", "Example of aws lambda container")
	c.AddFactory(build.NewAppExampleServiceFactory())
	return &c
}

File: app-container-example-nodex/src/container/AppExampleLambdaFunction.ts
import { CommandableLambdaFunction } from 'pip-services3-aws-nodex';
import { Descriptor } from 'pip-services3-commons-nodex';
import { AppExampleServiceFactory } from "../build";


export class AppExampleLambdaFunction extends CommandableLambdaFunction {
    public constructor() {
        super('app-example', 'Example of lambda function container');
        this._dependencyResolver.put(
            'controller',
            new Descriptor('app-example', 'controller', 'default', '*', '1.0'),
        );
        this._factories.add(new AppExampleServiceFactory());
    }
}

export const handler = new AppExampleLambdaFunction().getHandler();